orderer/common/multichannel/registrar.go

```diff
@@ -155,8 +160,57 @@ func NewRegistrar(
 func (r *Registrar) Initialize(consenters map[string]consensus.Consenter) {
 	r.consenters = consenters
 	existingChannels := r.ledgerFactory.ChannelIDs()
+	existingChannelsMap := map[string]struct{}{}
+	for _, channel := range existingChannels {
+		existingChannelsMap[channel] = struct{}{}
+	}
+
+	if r.config.ChannelParticipation.Enabled {
+		r.initializeFileRepos()
```

Comment:
This probably belongs in the `NewRegistrar`, returning an error, not panicking.

Reply:
Done